src/Cbor/Decode.elm

```diff
@@ -350,6 +368,13 @@ dict key value =
                 unsigned a |> D.andThen (\n -> D.loop ( n, [] ) finite)
 
 
+{-| Like 'array', but for heterogeneous dict.
+-}
+record : Decoder a -> Decoder a
+record =
+    Decoder (MajorType 5) << always << runDecoder
```

Comment:
Does this work? Don't you need some code to either drop the keys as you run into them, or to take a decoder for the keys and return a Dict?

Reply:
It works similarly to the `array` one, so `map` primitives are friends here. The decoding of key/value is to be done manually in that case. A `Dict` cannot be used for this is to cope precisely with heterogeneous structures which won't work with Elm's `Dict comparable a` type. This construct would also allow decoding records for which keys are not necessarily comparable, and heterogeneous too. For example, imagine a CBOR-map with 2 elements, the first one is a key:value from bool to int, and the second one a key:value from string to a list of bool. You would decode it as such:

```elm
type alias MyRecord = 
    { first : (Bool, Int)
    , second : (String, [Bool])
    }

D.decode <| D.record <|
    D.map2 MyRecord
        (D.pair D.bool D.int)
        (D.pair D.string (D.list D.bool))
````

We resort to representing each key:value pair as a 2-tuple.

Reply:
cool - thx